examples/siro_sandbox/controllers.py

```diff
@@ -0,0 +1,547 @@
+
+        if do_humanoidjoint_action:
+            humancontroller_base_user_input = [0, 0]
+            # temp keyboard controls to test humanoid controller
+            if gui_input.get_key(KeyNS.I):
+                # move in world-space x+ direction ("east")
+                humancontroller_base_user_input[0] += 1
+            if gui_input.get_key(KeyNS.K):
+                # move in world-space x- direction ("west")
+                humancontroller_base_user_input[0] -= 1
+
+            if self._hint_walk_dir:
+                humancontroller_base_user_input[0] += self._hint_walk_dir.x
+                humancontroller_base_user_input[1] += self._hint_walk_dir.z
+
+        action_names = []
+        action_args = {}
+        if do_humanoidjoint_action:
+            if True:
+                relative_pos = mn.Vector3(
+                    humancontroller_base_user_input[0],
+                    0,
+                    humancontroller_base_user_input[1],
+                )
+                # pose, root_trans = self._humanoid_controller.get_walk_pose(
+                #     relative_pos, distance_multiplier=1.0
+                # )
+                # humanoidjoint_action = self._humanoid_controller.vectorize_pose(
+                #     pose, root_trans
+                # )
+                # Use the controller
+                self._humanoid_controller.calculate_walk_pose(relative_pos)
+                humanoidjoint_action = self._humanoid_controller.get_pose()
+            else:
+                pass
+                # reference code
+                # humanoidjoint_action = self.get_random_joint_action()
+            action_names.append(humanoidjoint_name)
+            action_args.update(
+                {
+                    "agent_0_human_joints_trans": humanoidjoint_action,
+                }
+            )
+
+        return (
+            {"action": action_names, "action_args": action_args},
+            should_reset,
+            should_end,
+            {},
+        )
+
+
+class ControllerHelper:
+    def __init__(self, env, args, gui_input):
+        self.n_robots = len(env._sim.agents_mgr)
+        is_multi_agent = self.n_robots > 1
+
+        self._env = env
+
+        gui_controller: Controller = None
+        if args.humanoid_user_agent:
+            gui_controller = GuiHumanoidController(
+                0, is_multi_agent, gui_input, env, args.walk_pose_path
+            )
+        else:
+            gui_controller = GuiRobotController(0, is_multi_agent, gui_input)
+
+        self.controllers = []
+        self.n_robots = self.n_robots
+        self.all_hxs = [None for _ in range(self.n_robots)]
+        self.active_controllers = [0, 1]
+        self.controllers = [
+            gui_controller,
+            BaselinesController(
+                1,
+                is_multi_agent,
+                "habitat-baselines/habitat_baselines/config/rearrange/rl_hierarchical.yaml",
```

Comment:
`habitat-baselines/habitat_baselines/config/` config path prefix can be removed.